src/function/scalar/sequence/currval.cpp

```diff
@@ -0,0 +1,94 @@
+#include "duckdb/function/scalar/sequence_functions.hpp"
+
+#include "duckdb/catalog/catalog.hpp"
+#include "duckdb/catalog/catalog_entry/sequence_catalog_entry.hpp"
+#include "duckdb/common/exception.hpp"
+#include "duckdb/common/vector_operations/vector_operations.hpp"
+#include "duckdb/execution/expression_executor.hpp"
+#include "duckdb/catalog/catalog.hpp"
+#include "duckdb/planner/expression/bound_function_expression.hpp"
+#include "duckdb/transaction/transaction.hpp"
+#include "duckdb/common/vector_operations/unary_executor.hpp"
+
+namespace duckdb {
+
+struct CurrvalBindData : public FunctionData {
+	//! The client context for the function call
+	ClientContext &context;
+	SequenceCatalogEntry *sequence;
+
+	CurrvalBindData(ClientContext &context, SequenceCatalogEntry *sequence) : context(context), sequence(sequence) {
+	}
+
+	unique_ptr<FunctionData> Copy() override {
+		return make_unique<CurrvalBindData>(context, sequence);
+	}
+};
+
+static int64_t CurrSequenceValue(Transaction &transaction, SequenceCatalogEntry *seq) {
+	lock_guard<mutex> seqlock(seq->lock);
+	int64_t result;
+	if (seq->usage_count == 0u) {
+		throw SequenceException("currval: sequence has not been used yet.");
+	}
+	result = seq->counter - seq->increment;
+	seq->usage_count++; // should this be incremented?
+	transaction.sequence_usage[seq] = SequenceValue(seq->usage_count, seq->counter);
```

Comment:
The usage count only needs to be incremented on a change to the sequence to indicate that the change needs to be written to disk; I believe these two lines are not necessary.

Reply:
Ok, I have removed both lines